### PR TITLE
Fix implicit floating bool conversion

### PR DESCRIPTION
**Pull request by julienmalik**
clang reports a number of implicit float-to-bool conversions.

### CODE REVIEW

**Comment by julienmalik** [comment]
clang 3.9 warnings : 
* before : http://my.cdash.org/viewBuildError.php?type=1&buildid=949161
* after : http://my.cdash.org/viewBuildError.php?type=1&buildid=949171

**Comment by julienmalik** [comment]
Do NOT merge as-is since it is based on #753

File: src/lib/openjp2/tcd.c [hunk 1]
```diff
@@ -487,7 +487,7 @@ OPJ_BOOL opj_tcd_rateallocate(  opj_tcd_t *tcd,
                 OPJ_FLOAT64 lo = min;
                 OPJ_FLOAT64 hi = max;
                 OPJ_BOOL success = OPJ_FALSE;
-                OPJ_UINT32 maxlen = tcd_tcp->rates[layno] ? opj_uint_min(((OPJ_UINT32) ceil(tcd_tcp->rates[layno])), len) : len;
+                OPJ_UINT32 maxlen = tcd_tcp->rates[layno] > 0 ? opj_uint_min(((OPJ_UINT32) ceil(tcd_tcp->rates[layno])), len) : len;
```
Comment (mayeut) [inline src/lib/openjp2/tcd.c]: Wouldn't `tcd_tcp->rates[layno] > 0.0f` be more explicit ?

Comment (julienmalik) [inline src/lib/openjp2/tcd.c]: sure

**Comment by julienmalik** [comment]
With or without the warning fixes :

http://my.cdash.org/index.php?project=OPENJPEG&date=2016-04-27&filtercount=2&showfilters=1&filtercombine=or&field1=buildname/string&compare1=63&value1=Ubuntu12.04-clang3.9.0-x86_64-master-pr752&field2=buildname/string&compare2=63&value2=Ubuntu12.04-clang3.9.0-x86_64-master-pr753

**Comment by mayeut** [comment]
@julienmalik, could you refactor this PR following #753 merge

Thanks

**Pull request merged**